### ISSUE TITLE
Visual Composer and Ajax issue

### ISSUE THREAD

**Issue by sstruemph** [WordImpress/Google-Maps-Builder#153]
Map isn't loading with Visual Composer and the VC Tabs moduble (Tabbed Content) without page refresh.

To see the issue on the customer's site go to: http://www.tonicpit.com/kokoda/residences/#grid. Then click on "View Map". The map isn't drawing. If you refresh or go directly to the map tab then the map loads: http://www.tonicpit.com/kokoda/residences/#map.

![screenshot-www tonicpit](https://cloud.githubusercontent.com/assets/3521793/13362704/89180c3a-dc8b-11e5-98d0-ebfc5b4c628c.jpg)

**Comment by DevinWalker** [comment]
Here's the fix for this:

```
//VC Tabs:
        $('.vc_tta-tab').on('click', function (e) {
            setTimeout(function (){
                // Something you want delayed.
                var panel = $('.vc_tta-panels-container').get(0);
                gmb.load_hidden_map(panel);
            }, 400);
        });
``` 

I just need to get it into both plugins now.

**Comment by Shelob9** [comment]
Does Visual Composer have a tab loaded event?

**Comment by DevinWalker** [comment]
@Shelob9 I tried to dig into it but they only ship minified versions of the js so it was hard to find any... I doubt it based on that. I did however, get it to work.

**Comment by DevinWalker** [comment]
... but it's kinda hacky with the usage of `setTImeout`

**Comment by DevinWalker** [comment]
`jQuery('.vc_tta-tabs a').on('show.vc.tab', function(){
 google.maps.event.trigger(window, 'resize', {});
});` from: https://wordpress.org/support/topic/map-showing-graybox-inside-a-tab-visual-composer?replies=4#post-8642549

**Comment by DevinWalker** [comment]
Fixed here 3bba6819a13542a64398abdc18c0e569fc8c92cf

**Comment by imrantushar** [comment]
@DevinWalker Thank You :+1:
